gosh list versions on an empty config file and no deployment repo

Running the command with an empty config file and on a repo that contains no deployments

docker run -v $PWD:/workdir -v $PWD/config.yml:/root/.gosh/config.yml gosh list versions

Output:

panic: interface conversion: interface {} is nil, not map[string]interface {}
goroutine 1 [running]:
gosh/util.InitializeConfig()
        /__w/gosh/gosh/util/config.go:62 +0xad9
github.com/spf13/cobra.(*Command).preRun(0xff6be0)
        /go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:882 +0x49
github.com/spf13/cobra.(*Command).execute(0xff6be0, 0xc00018bac0, 0x2, 0x2, 0xff6be0, 0xc00018bac0)
        /go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:818 +0x14f
github.com/spf13/cobra.(*Command).ExecuteC(0xff6e60, 0x200000003, 0xc000000180, 0xc00011df78)
        /go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:960 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
        /go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:897
gosh/cmd.Execute(0xffffffff, 0xc000096058)
        /__w/gosh/gosh/cmd/root.go:34 +0x2d
main.main()
        /__w/gosh/gosh/main.go:9 +0x25

--- a/.commitlintrc.yml
+++ b/.commitlintrc.yml
@@ -14,6 +14,7 @@ rules:
       - perf
       - test
       - revert
+  body-max-line-length: 500
 help: |
   **Possible types**:
   `chore`:    Change build process, tooling or dependencies.

--- a/.commitlintrc.yml
+++ b/.commitlintrc.yml
@@ -14,7 +14,7 @@ rules:
       - perf
       - test
       - revert
-  body-max-line-length: 500
+  body-max-line-length: [1, 'always', 500]
 help: |
   **Possible types**:
   `chore`:    Change build process, tooling or dependencies.

--- a/util/config.go
+++ b/util/config.go
@@ -8,9 +8,8 @@ import (
 )
 
 const (
-	GoshConfigDir         = ".gosh"
-	GoshDefaultConfigFile = "$HOME/" + GoshConfigDir + "/" + GoshConfigFile
-	GoshConfigFile        = "config.yml"
+	GoshConfigDir  = ".gosh"
+	GoshConfigFile = "config.yml"
 )
 
 type DeploymentRepository struct {
@@ -49,25 +48,34 @@ func InitializeConfig() {
 	}
 	viper.SetEnvPrefix("GOSH")
 	viper.AutomaticEnv()
-	viper.SetConfigFile(os.ExpandEnv(GoshDefaultConfigFile))
-	if err := viper.ReadInConfig(); err == nil {
-		if err = viper.MergeConfigMap(v.AllSettings()); err != nil {
-			log.Fatal(err, "Error merging configuration")
+	homedir, _ := os.UserHomeDir()
+	configFile := filepath.Join(homedir, GoshConfigDir, GoshConfigFile)
+	viper.SetConfigFile(configFile)
+	if _, err := os.Stat(configFile); err == nil {
+		if err := viper.ReadInConfig(); err == nil {
+			if err = viper.MergeConfigMap(v.AllSettings()); err != nil {
+				log.Fatal(err, "Error merging configuration")
+			}
+		} else {
+			if _, ok := err.(viper.ConfigFileNotFoundError); !ok {
+				log.Fatal(err, "Could not read configuration")
+			}
 		}
-	} else {
-		if _, ok := err.(viper.ConfigFileNotFoundError); !ok {
-			log.Fatal(err, "Could not read configuration")
+		if viper.IsSet("deploymentrepository") {
+			Config.DeploymentRepository = newDeploymentRepository(viper.Get("deploymentrepository").(map[string]interface{}))
 		}
+		Config.ArtifactRepositories = make(map[string]map[string]string, 0)
+		if viper.IsSet("artifactrepositories") {
+			settings := viper.Get("artifactrepositories").(map[string]interface{})
+			for name, urls := range settings {
+				result := map[string]string{}
+				for k, v := range urls.(map[string]interface{}) {
+					result[k] = v.(string)
+				}
+				Config.ArtifactRepositories[name] = result
+			}
+		}
+		log.Debugf("Loaded configuration %+v", Config)
 	}
-	Config.DeploymentRepository = newDeploymentRepository(viper.Get("deploymentrepository").(map[string]interface{}))
-	Config.ArtifactRepositories = make(map[string]map[string]string, 0)
-	settings := viper.Get("artifactrepositories").(map[string]interface{})
-	for name, urls := range settings {
-		result := map[string]string{}
-		for k, v := range urls.(map[string]interface{}) {
-			result[k] = v.(string)
-		}
-		Config.ArtifactRepositories[name] = result
-	}
-	log.Debugf("Loaded configuration %+v", Config)
+
 }

--- a/util/config_test.go
+++ b/util/config_test.go
@@ -1,0 +1,29 @@
+package util
+
+import (
+	"github.com/Flaque/filet"
+	"github.com/stretchr/testify/suite"
+	"os"
+	"testing"
+)
+
+type ConfigTestSuite struct {
+	suite.Suite
+}
+
+func (suite *ConfigTestSuite) SetupSuite() {
+}
+
+func (suite *ConfigTestSuite) TestInitializeConfig_NoConfig() {
+	os.Setenv("HOME", filet.TmpDir(suite.T(), ""))
+	InitializeConfig()
+	//r := suite.Require()
+}
+
+func (suite *ConfigTestSuite) TearDownSuite() {
+	filet.CleanUp(suite.T())
+}
+
+func TestConfigTestSuite(t *testing.T) {
+	suite.Run(t, new(ConfigTestSuite))
+}

